roles/publish-charm/defaults/main.yaml

```diff
@@ -0,0 +1,5 @@
+needs_charm_build: false
+charm_build_name: null
+build_type: charmcraft
+publish_charm: false
+publish_channel: latest/edge
```

Comment:
I'm not sure if it's needed here (I can never work out Ansible's rules on variables), but it might be useful to add a default for `charmcraft_channel` here as `2.0/stable` (or whatever is currently being used on the sunbeam charms).  This would be for the charm-build role (which default `charmcraft_channel` to `1.5/stable` for the legacy backports.

Reply:
Added charmcraft_channel in defaults.

Reply:
This is nothing to change in this PR, but I do wonder if we the 2.0/stable is the right channel to target for the default value. Certainly we should default to something (we've been bitten by not pinning to a specific branch or lack thereof for other libraries/tools in the system). 2.x/stable might be a more aggressive yet sufficiently pinned version, *however* it depends on what commitment we may be able to get on stability of the 2. major version here. Typically, I would expect breaking changes to come at a major version bump, but that hasn't historically been the case.

Ultimately, I think 2.0/stable is fine for now.